python/src/extensions.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright © 2023-2024 Apple Inc.
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "mlx/extensions.h"
+#include "mlx/ops.h"
+#include "python/src/utils.h"
+
+namespace py = pybind11;
+using namespace py::literals;
+using namespace mlx::core;
+
+void init_extensions(py::module_& parent_module) {
+  py::options options;
+  options.disable_function_signatures();
+
+  auto m = parent_module.def_submodule("ext", "mlx.core.ext: fast operations");
+
+  m.def(
+      "rope",
+      [](const array& x,
+         int dims,
+         bool traditional,
+         float base,
+         float scale,
+         int offset,
+         const StreamOrDevice& s /* = {} */) {
+        return ext::rope(x, dims, traditional, base, scale, offset, s);
+      },
+      "x"_a,
+      "dims"_a,
+      "traditional"_a,
+      "base"_a,
+      "scale"_a,
+      "offset"_a,
```

Comment:
Do you think we should make the above keyword only? It would be verbose but error free...

Reply:
I think so, yes.